lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -134,6 +134,28 @@ def create(path)
       @zk.create(path, ignore: :node_exists)
     end
 
+    # get with retry to reduce failure when network connectivity is flaky or ZK is overloaded
+    def get(path, opts = {})
+      retry_limit = opts[:retry_limit] || 3
+      opts.delete(:retry_limit)
```

Comment:
hmm, actually, why do we need to delete `:retry_limit`? it will be still used for further fetch zk path, right?

Reply:
Here I hesitate to pass some parameters to zookeeper client API while zookeeper doesn't understand what it means. I only want to pass along the rest of parameters specified by caller. And I'm new to Ruby, and couldn't find a better way to express that after some research. If you know, please let me know.